excel/pkg/graph/worksheet.go

```diff
@@ -105,6 +110,96 @@ func GetWorksheetData(ctx context.Context, c *msgraphsdkgo.GraphServiceClient, w
 	return data, usedRange, nil
 }
 
+func GetWorksheetTables(ctx context.Context, c *msgraphsdkgo.GraphServiceClient, workbookID, worksheetID string) ([]Table, error) {
+	drive, err := c.Me().Drive().Get(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	result, err := c.Drives().ByDriveId(util.Deref(drive.GetId())).Items().ByDriveItemId(workbookID).Workbook().Worksheets().ByWorkbookWorksheetId(worksheetID).Tables().Get(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	var tables []Table
+	for _, table := range result.GetValue() {
+		tables = append(tables, Table{ID: util.Deref(table.GetId()), Name: util.Deref(table.GetName())})
+	}
+
+	return tables, nil
+}
+
+func GetColumnIDFromName(ctx context.Context, c *msgraphsdkgo.GraphServiceClient, workbookID, worksheetID, tableID, columnName string) (string, error) {
+	drive, err := c.Me().Drive().Get(ctx, nil)
+	if err != nil {
+		return "", err
+	}
+	result, err := c.Drives().ByDriveId(util.Deref(drive.GetId())).Items().ByDriveItemId(workbookID).Workbook().Tables().ByWorkbookTableId(tableID).Columns().Get(ctx, nil)
+	if err != nil {
+		return "", err
+	}
+	for _, column := range result.GetValue() {
+		rangeColumnName := util.Deref(column.GetName())
+		if rangeColumnName == columnName {
+			return util.Deref(column.GetId()), nil
+		}
+	}
+	return "", errors.New("column not found")
+}
+
+func FilterWorksheetData(ctx context.Context, c *msgraphsdkgo.GraphServiceClient, workbookID, worksheetID, tableID, columnID string, filterValues []string) ([][]any, models.WorkbookRangeable, error) {
+	criteria := map[string]any{
+		"criteria": map[string]any{
+			"filterOn": "Values",
+			"Values":   filterValues,
+		},
+	}
+
+	bodyJSON, err := json.Marshal(criteria)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, fmt.Sprintf("https://graph.microsoft.com/v1.0/me/drive/items/%s/workbook/worksheets/%s/tables/%s/columns/%s/filter/apply", workbookID, worksheetID, tableID, columnID), strings.NewReader(string(bodyJSON)))
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer"+os.Getenv(global.CredentialEnv))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to send request: %w", err)
+	}
```

Comment:
Even if the body is empty, I think it still needs to be closed.
```suggestion
	}
	defer resp.Body.Close()
```